### PR TITLE
TEST: increase threshold for memory usage on tests

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

ref https://github.com/intel/scikit-learn-intelex/pull/2107

This PR increases the threshold for the tests that check for memory usage increases to 25% of the input data size, as those tests were previously producing failures on some CI runs in that other PR.

Regardless of whether there are or aren't any memory leaks, it is not guaranteed that this condition would be met, since a memory allocator is not required to return all memory to the OS after freeing it, but this increase should allow the CI jobs to pass until a better solution is found.

Note that the PR contains an unrelated change - this is coming from the precommit hook, which required that change to go into the PR.

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Review by samir-nasibli — COMMENTED** [review]
I think better to use this specific value for specific cases

**Comment by david-cortes-intel** [comment]
> I think better to use this specific value for specific cases

Changed it use 25% for pandas inputs, while leaving it at 15% for other types of inputs.

**Comment by Alexsandruss** [comment]
`Check PR Checklist / Close all checkboxes before moving from draft`

**Review by samir-nasibli — APPROVED** [review]
Looks good! Assuming green internal CI as well.
For me make sense also cover with ticket further investigation the reason of such number increase

File: sklearnex/tests/test_memory_usage.py [hunk 1]
```diff
@@ -226,15 +227,20 @@ def _kfold_function_template(estimator, dataframe, data_shape, queue=None, func=
     else:
         name = estimator.__name__
 
+    threshold = (
+        EXTRA_MEMORY_THRESHOLD_PANDAS
+        if isinstance(dataframe, pd.DataFrame)
```
Comment (samir-nasibli) [inline sklearnex/tests/test_memory_usage.py:232]: could it be `pd.Series` somehow?

Comment (david-cortes-intel) [reply sklearnex/tests/test_memory_usage.py]: Thanks for pointing this out. It made me realize that I was actually checking the wrong parameter ("X" here would be the dataframe, whereas this "dataframe" is a string) - fixed it now.
